server/src/main/java/com/microsoft/java/bs/core/internal/gradle/actions/GetSourceSetsAction.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.java.bs.core.internal.gradle.actions;
+
+import com.microsoft.java.bs.gradle.model.GradleSourceSet;
+import com.microsoft.java.bs.gradle.model.GradleSourceSets;
+import com.microsoft.java.bs.gradle.model.impl.DefaultGradleSourceSets;
+import org.gradle.tooling.BuildAction;
+import org.gradle.tooling.BuildController;
+import org.gradle.tooling.model.gradle.GradleBuild;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.stream.Collectors;
+
+/**
+ * {@link BuildAction} that retrieves {@link GradleSourceSets} from a Gradle build,
+ * handling both normal and composite builds.
+ */
+public class GetSourceSetsAction implements BuildAction<GradleSourceSets> {
+
+  /**
+   * Executes the build action and retrieves source sets from the Gradle build.
+   *
+   * @return A {@link DefaultGradleSourceSets} object containing all retrieved source sets.
+   */
+  @Override
+  public GradleSourceSets execute(BuildController buildController) {
+    Map<String, List<GradleSourceSet>> sourceSets = new HashMap<>();
```

Comment:
You could make use of `var`, since it's java 17

```suggestion
    var sourceSets = new HashMap<String, List<GradleSourceSet>>();
```

Reply:
Done ✅